plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,390 @@
+        choices: ['encrypt', 'sign', 'auth', 'cert']
+    name:
+        description:
+            - Specifies a name for the key.
+        type: str
+    comment:
+        description:
+            - Specifies a comment for the key.
+        type: str
+    email:
+        description:
+            - Specifies an email for the key.
+        type: str
+    passphrase:
+        description:
+            - Passphrase used to decrypt an existing private key or encrypt a newly generated private key.
+            - If O(state) is V(absent), this parameter is ignored. 
+        type: str
+    fingerprints:
+        description:
+            - Specifies keys to match against.
+            - Provided fingerprints will take priority over user-id "V(name) (V(comment)) <V(email)>".
+            - If O(state) is V(absent), keys with the provided fingerprints will be deleted if found 
+        type: list[str]
+    keyserver:
+        description:
+            - Specifies keyserver to upload key to.
+            - If O(state) is V(absent), this parameter will be ignored.
+        type: str
+    transient_key:
+        description:
+            - Allows key generation to use a faster, but less secure random number generator.
+        type: bool
+        default: False
+    return_fingerprints:
+        description:
+            - Allows for the return of fingerprint(s) for newly created or deleted keys(s)
+        type: bool
+        default: False
+'''
+
+EXAMPLES = '''
+- name: Generate the default GPG keypair (Ed25519)
+  community.crypto.gpg_keypair:
+
+- name: Generate the default GPG keypair with a passphrase
+  community.crypto.gpg_keypair:
+    passphrase: super_secret_password
+
+- name: Generate a RSA GPG keypair with the default RSA size (2048 bits)
+  community.crypto.gpg_keypair:
+    key_type: RSA
+
+- name: Generate a RSA GPG keypair with custom size (4096 bits)
+  community.crypto.gpg_keypair:
+    key_type: RSA
+    key_length: 4096
+
+- name: Generate an ECC GPG keypair 
+  community.crypto.gpg_keypair:
+    key_type: EDDSA
+    key_curve: cv25519
+
+- name: Generate a GPG keypair and with a subkey:
+  community.crypto.gpg_keypair:
+    subkey_type: ECDH
+    subkey_curve: cv25519
+
+- name: Generate a GPG keypair with custom user-id:
+  community.crypto.gpg_keypair:
+    name: Your Name
+    comment: Interesting comment.
+    email: example@email.com
+
+- name: Generate a GPG keypair and return fingerprint of new key
+  community.crypto.gpg_keypair:
+    return_fingerprints: true
+  register: gpg_keys
+
+- name: Delete GPG keypair(s) matching a specified user-id:
+  community.crypto.gpg_keypair:
+    state: abscent
+    name: Your Name
+    comment: Interesting comment.
+    email: example@email.com
+
+- name: Delete GPG keypair(s) matching a specified fingerprint:
+  community.crypto.gpg_keypair:
+    state: abscent
+    fingerprints:
+      - ABC123...
+
+'''
+
+RETURN = '''
+size:
+    description: Size (in bits) of the SSH private key.
```

Comment:
SSH → GPG? :)

Reply:
Thought I deleted the size return value entirely. Whoops